mlflow/store/artifact/optimized_s3_artifact_repo.py

```diff
@@ -65,9 +70,27 @@ def _get_region_name(self):
             s3_endpoint_url=self._s3_endpoint_url,
         )
         try:
-            return temp_client.head_bucket(Bucket=self.bucket)["BucketRegion"]
+            head_bucket_resp = temp_client.head_bucket(Bucket=self.bucket)
+            if _BUCKET_REGION in head_bucket_resp:
+                return head_bucket_resp[_BUCKET_REGION]
+            if (
+                _RESPONSE_METADATA in head_bucket_resp
+                and _HTTP_HEADERS in head_bucket_resp[_RESPONSE_METADATA]
+                and _HTTP_HEADER_BUCKET_REGION
+                in head_bucket_resp[_RESPONSE_METADATA][_HTTP_HEADERS]
+            ):
```

Comment:
nit: can we leave a comment explaining what these various branches represent?

Reply:
sure.